manual_test_guides.md

```diff
@@ -0,0 +1,52 @@
+O3DE Manual Test Guides
+--
+The following test guides are recommended to be used for manual test verification of any changes made to their corresponding features, components and systems. Due to the complexity of the O3DE product and tools it has been determined that automated tests alone are not sufficient to fully mitigate the risk of escaped defects in submitted changes. The purpose of these manual test workflows are to serve as a supplement to the Automated Review test system and have been proven effective at uncovering defects not found through automated tests.
```

Comment:
I think this could use a soft-recommendation of how often SIGs should manually walk through their guides, and potentially also clarification that SIG-Testing does not organize resources to do so on their behalf.

Reply:
Added a note about this in the latest commit.